File: c-sharp-design-patterns/OOP Principles/Abstraction/BadEmailService.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace c_sharp_design_patterns.OOP_Principles.Abstraction
{
    public class BadEmailService
    {
        public void Connect()
        {
            Console.WriteLine("Connecting to email service.");
        }

        public void Authenticate()
        {
            Console.WriteLine("Authenticating user.");
        }

        public void SendEmail()
        {
            Console.WriteLine("Sending email.");
        }

        public void Disconnect()
        {
            Console.WriteLine("Disconnecting from email service.");
        }

    }
}

File: OOP_Principles/Abstraction/BadEmailService.java
package OOP_Principles.Abstraction;

public class BadEmailService {

  public void Connect() {
    System.out.println("Connecting to email service.");
  }

  public void Authenticate() {
    System.out.println("Authenticating user.");
  }

  public void SendEmail() {
    System.out.println("Sending email.");
  }

  public void Disconnect() {
    System.out.println("Disconnecting from email service.");
  }
}
